File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/interfaces/IController.java
package com.sergey.pisarev.interfaces;

public interface IController {
    void onStop();

    void showProgram(String text);

    void showFrame(String frame);

    void showError(String error);

    void showCaretBoxOnCycleStart(int number, StringBuffer frame);

    void setZooming(double zooming);

    void getCoordinateCanvas(double x,double z);

    void getCoordinateFrame(double x,double z);

    void showCaretBoxOnCanvasClick(int number, StringBuffer frame);

    void showSaveAlert();
}

File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/interfaces/IController.kt
package com.sergey.pisarev.interfaces

import java.lang.StringBuffer

interface IController {
    fun onStop()
    fun showProgram(text: String?)
    fun showFrame(frame: String?)
    fun showError(error: String?)
    fun showCaretBoxOnCycleStart(number: Int, frame: StringBuilder?)
    fun setZooming(zooming: Double)
    fun getCoordinateCanvas(x: Double, z: Double)
    fun getCoordinateFrame(x: Double, z: Double)
    fun showCaretBoxOnCanvasClick(number: Int, frame: StringBuilder?)
    fun showSaveAlert()
}